### ISSUE TITLE
Filtering - ignore all non-alphanumberic characters

### ISSUE THREAD

**Issue by mcsmithers** [Mottie/tablesorter#1602]
Hi everyone,

I am trying to play with the examples to see if I can make the filter search for an entry that has non-alphanumeric characters, but doesn't search them exactly.  I thought a wild search or a fuzzy one might work, however, I can't seem to quite get them to work.  

Here's an example of some of the data:
`ZNS - TDL - Cruise Deals - CPC - Display`
`ZNS - New Cars - Direct - CPC - Display`
`Medigap.com_dCPC_Display`
`ZNSNewCars - Mobile - US 7203 [WLp Testing Smartbid GIF test] 7203_2092`

And searches would be:
`ZNS Cruise Desktop`
`ZNS New Car US scaled`
`medigap`
`Testing Even mobile US`

Do you have any suggestions on how to use the tool for this?  I am getting it to work somewhat, but if you have any suggestions on what kind of filtering or options to use more, I am all ears. I get no matches with wild or  fuzzy and probably from the dashes.

```
$('#table table')
            .tablesorter({
                // this option only adds a table class name "tablesorter-{theme}"
                theme: 'default',
                widgets: ['sortTbody', 'filter', 'zebra'],
                widgetOptions: {
                    filter_external: '.search',
                    filter_columnFilters: false,
                    resizable: true,
                    filter_defaultFilter: {
                        all: '*{q}',  // also tried ~{q}
                    },
                    filter_ignoreCase: true,
                    filter_external : '.search',
                    filter_saveFilters : true,
                    filter_reset: '.reset',
                    sortTbody_lockHead: true,
                    sortTbody_primaryRow: '.summary',
                    sortTbody_sortRows: true,
                    zebra: ["even", "odd"],
                    usNumberFormat: true,
                    // include child row content while filtering the second demo table
                    filter_childRows: true
                }
            });
```

Thank you so much for your patience and help in this wonderful tool.

**Comment by Mottie** [comment]
Hi @mcsmithers!

You're right about the dashes. You'll need to turn off the range filter search to get thinks to work as expected.

Set the [`filter_excludeFilter` option](https://mottie.github.io/tablesorter/docs/example-widget-filter.html#filter-excludefilter) for the column:

```js
filter_defaultFilter: {
  ".tablesorter-header": "~{q}"
},
filter_excludeFilter: {
  ".tablesorter-header": "range"
}
```

Also, please use a css selector (`.tablesorter-header` or `th`) in both of these options; using `all` won't work as expected.

**Comment by mcsmithers** [comment]
Okay, so doing a filter for all the columns [like this example](http://jsfiddle.net/Mottie/abkNM/2566/) with this option won't be possible then?

**Comment by Mottie** [comment]
Oh, yes. An external filter will automatically ignore ranges so you don't need to set the `filter_excludeFilter`. And if I recall correctly, I don't think the `filter_defaultFilter` works for external search inputs.

**Comment by Mottie** [comment]
Ok, I totally forgot I did implement it! I had to set up a demo to test it...

If you set the `filter_defaultFilter` option to use the total number of columns as an index, it'll start working! The generic `th` wouldn't work because there is no "any" column.

In [this demo](http://jsfiddle.net/Mottie/ps571q2m/) there are two columns, so set:

```js
filter_defaultFilter: {
	2: '~{q}'
}
```

It's because the filter sets the "any" column to the last column plus one, using a zero-based index.

If you're still having issues with the range filter, I commented out this line in the demo:

```js
delete $.tablesorter.filter.types.range;
```

which can be used to completely remove the range filter.
